subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompiler.java

```diff
@@ -47,12 +50,18 @@ public WorkResult execute(TwirlCompileSpec spec) {
             ClassLoader cl = getClass().getClassLoader();
             ScalaMethod compile = adapter.getCompileMethod(cl);
             Iterable<RelativeFile> sources = spec.getSources();
-            for (RelativeFile sourceFile : sources) {
-                Object result = compile.invoke(adapter.createCompileParameters(cl, sourceFile.getFile(), sourceFile.getBaseDir(), spec.getDestinationDir(), spec.getDefaultImports()));
-                ScalaOptionInvocationWrapper<File> maybeFile = new ScalaOptionInvocationWrapper<File>(result);
-                if (maybeFile.isDefined()) {
-                    outputFiles.add(maybeFile.get());
+            for (final RelativeFile sourceFile : sources) {
+                TwirlTemplateFormat format = findTemplateFormat(spec, sourceFile.getFile());
+                // Only generate files if a format for this file's extension is found
```

Comment:
Is this better behaviour? Maybe we should fail for an input file that we can't handle, rather than silently ignoring it?